### DISCUSSION TITLE
Role permissions and task instance data unexpectedly disappearing

### DISCUSSION THREAD

**Discussion by imamdigmi** [apache/airflow#33437] [General]
After upgrading Airflow from version 2.3.3 to 2.6.3, I've encountered two significant issues related to role permissions and task instance data disappearing without manual intervention:

1. Role Permissions:
I created a custom role and assigned a specific DAG ID to this role.
The assignment was successful initially.
However, after a few minutes and upon refreshing the role permission page, the assigned permissions disappeared.

2. Task Instance Data:
I modified a DAG's schedule interval from @daily to hourly (0 */1 * * *) and adjusted the start date.
After testing and confirming no errors, I deployed the updated DAG and deleted the old version of the DAG using the web UI.
On unpausing the new DAG, it initiated its catchup runs as expected, starting from 2023-08-15.
A few minutes into the execution, the DAG Run status abruptly changed to 'success', but the task instance data was blank/missing.
This issue wasn't limited to the modified DAG; other DAGs, which I hadn't changed, also exhibited the same behavior.

![airflow-issue-1](https://github.com/apache/airflow/assets/10998240/19aae766-42c2-420a-a94b-216ff45d731c)
![airflow-issue-2](https://github.com/apache/airflow/assets/10998240/412d989c-07dd-473b-9167-f2f78d364682)
![airflow-issue-3](https://github.com/apache/airflow/assets/10998240/b21e59e6-3b91-4258-8821-e678655f746c)

I even check on the metadata database, all task instance related to the DAGs are dissapeared using this query:
`select * from task_instance where dag_id = '....' and run_id = 'scheduled__2023-08-13T17:00:00+00:00' order by run_id desc`

I'm seeking guidance on how to debug these issues further and any potential solutions or workarounds to address them. Any insights or recommendations from the community would be greatly appreciated.

**Comment by potiuk** [comment]
1. I think this is another instance of https://github.com/apache/airflow/issues/32839 which still have not found the right solution.
2. I think you need to explain more details on how you uploaded a new DAG and deployed a new one, I am not sure I follow from your description. Ideally a reproducible case with step-by-step what you have done and some of your findings would be great (and ALWAYS , ALWAYS -> one entry, one issue/discussion. Never try to put two issues into one discussion, this is not going to end well and makes everyone life's harder.

**Reply by potiuk** [reply]
Creating a separate issue for 2) would be a good start if you can provide enough reproducibility and details.
